test/test_config/test_configmanager.cc

```diff
@@ -0,0 +1,134 @@
+#include <array>
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include <uuid/uuid.h>
+#include <fstream>
+
+#include <config_manager.h>
+#include <config/config_generator.h>
+
+using namespace zmm;
+using namespace mxml;
+
+class ConfigManagerTest : public ::testing::Test {
+
+ public:
+
+  ConfigManagerTest() {};
+
+  virtual ~ConfigManagerTest() {};
+
+  virtual void SetUp() {
+    gerberaDir = createTempPath();
```

Comment:
`ConfigManager` requires a filesystem file to be initialized.  It also checks the validity of the script files referenced in config.

Initially was using **boost** library to generate temporary dir and files, but its a bit heavy weight to include just for testing.